Prb.Radiators.Wpf/MainWindow.xaml.cs

```diff
@@ -20,10 +20,159 @@ namespace Prb.Radiators.Wpf
+            panLower.IsEnabled = false;
+        }
+
+        private void CalculateRoomVolume()
+        {
+            double lengthRoom = double.Parse(txtRoomLength.Text);
+            double widthRoom = double.Parse(txtRoomWidth.Text);
+            double heightRoom = double.Parse(txtRoomHeight.Text);
+
+            double volume = lengthRoom * widthRoom * heightRoom / 1000000;
+            double watts = volume * 50;
+
+            lblRoomValue.Content = $"{volume: n2}".ToString();  //Alles afronden naar 0.00
+            lblRoomWatts.Content = watts.ToString();
+
+        }
+
+        private void CalculateRoomWatts()
+        {
+            lblRoomWatts.Text= $"{roomVolume * 50} W";
+        }
+
+        private void Window_Loaded(object sender, RoutedEventArgs e)
+        {
+            panTop.IsEnabled = true;
+            panLower.IsEnabled = false;       //panBottom = CS0229 error
+            FillComboBoxes();
+            
+            
+        }
+
+
+        private void BtnCalculateVolumeAndWatts_Click(object sender, RoutedEventArgs e)
+        {
+            CalculateRoomVolume();
+
+            panTop.IsEnabled = false;
+            panLower.IsEnabled = true;
+        }
+
+        private void BtnCalculateRadiatorWatts_Click(object sender, RoutedEventArgs e)
+        {
+            double heightRadiator = double.Parse(cmbRadiatorHeight.SelectedValue.ToString());
+            double width = double.Parse(cmbRadiatorWidth.SelectedValue.ToString());
+            int panels = int.Parse(cmbRadiatorPanels.SelectedValue.ToString());
+
+            double watts = heightRadiator * width * panels * 0.15;
+            lblRadiatorWatts.Content = watts.ToString();
+            btnAddRadiatorHeight.Visibility = Visibility.Visible;
+        }
+
+        private void CmbRadiatorHeight_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            lblRadiatorWatts.Content = "";
+            btnAddRadiatorHeight.Visibility = Visibility.Collapsed;
+
+        }
+
+        private void CmbRadiatorWidth_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            lblRadiatorWatts.Content = "";
+            btnAddRadiatorHeight.Visibility = Visibility.Collapsed;
+        }
+
+        private void CmbRadiatorPanels_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            lblRadiatorWatts.Content = "";
+            btnAddRadiatorHeight.Visibility = Visibility.Collapsed;
+        }
+
+        private void BtnAddRadiatorHeight_Click(object sender, RoutedEventArgs e)
+        {
+            double height = double.Parse(cmbRadiatorHeight.SelectedValue.ToString());
+            double width = double.Parse(cmbRadiatorWidth.SelectedValue.ToString());
+            int panels = int.Parse(cmbRadiatorPanels.SelectedValue.ToString());
+
+            double watts = height * width * panels * 0.15;
+
+            lstRadiators.Items.Add($"Radiator ({height} x {width} x {panels}) - {watts} W");
```

Comment:
Ook het opstellen van de tekst kan in een aparte methode die een string teruggeeft.